### ISSUE TITLE
Target quadratic objectives when calling SCS

### ISSUE THREAD

**Issue by bodono** [cvxpy/cvxpy#1605]
SCS has native support for [quadratic objectives](https://www.cvxgrp.org/scs/index.html) now. If a problem has a quadratic objective then solving it directly is often much, much faster than converting to a second-order cone program.

**Comment by rileyjmurray** [comment]
Very much in favor of this. It would require creating a new standard form. Right now we have cone programs (linear) and QPs (with only LP-representable constraints). Perhaps a GSoC project, @akshayka @SteveDiamond, @bstellato?

**Comment by jakeknigge** [comment]
Hi @rileyjmurray and @SteveDiamond, I'd traded a couple emails with @bodono about contributing to SCS (I’ve benefitted from using CVX{PY, R, ""} [where CVX"" = CVX] and the related ecosystem for years and I'd like to start paying my CVXGRP dues). He suggested that this issue could be an interesting place to start—and I agree! It looks like a fun one, touching a variety of places in the stack. Let me know what you have in mind and how we can proceed. Thanks!

**Comment by SteveDiamond** [comment]
Hi @jakeknigge ! Can you join the discord (https://discord.gg/4urRQeGBCr) and reach out to us there? Then we can discuss approaches in more detail.

**Comment by SteveDiamond** [comment]
@jakeknigge could you email me? I'd like to chat about how things are going with this, since we have a new solver that requires this interface.

**Comment by SteveDiamond** [comment]
Here is an outline for how this might be implemented:

1. Modify ``ConeMatrixStuffing`` so that it can take a quadratic objective.
2. Modify ``ParamConeProg`` so that it can take a quadratic objective (like ``ParamQuadProg``).
3. Add a ``quad_objective`` boolean flag to ``ConicSolver``. This will be false by default and true for SCS and other conic solvers that support quadratic objectives.
4. Modify ``dcp2cone`` so that it will preserve/canonicalize quadratic objectives if the ``quad_objective`` flag is true.

Basically I think it's better to extend the definition of a conic solver to include a quadratic objective, rather than write a new set of reductions. We'll have to be careful that there isn't a performance regression for the standard (non-quadratic objective) conic reduction path, but we can worry about that after the basic functionality is working.

If we go this route, then the QP reduction path might become superfluous. We can think about that later.
